tensorbay/client/segment.py

```diff
@@ -652,6 +661,55 @@ class FusionSegmentClient(SegmentClientBase):
     def __init__(self, name: str, data_client: "FusionDatasetClient") -> None:
         super().__init__(name, data_client)
 
+    @staticmethod
+    def _wrap_callback_info(
+        callback_info: Dict[str, Any], sensor_name: str, frame_id: str, data: Data
+    ) -> None:
+        callback_info["sensorName"] = sensor_name
+        callback_info["frameId"] = frame_id
+
+        if hasattr(data, "timestamp"):
+            callback_info["timestamp"] = data.timestamp
+
+        if data.label:
+            callback_info["label"] = data.label.dumps()
+
+    def _generate_upload_data(
+        self,
+        source_frames: Iterator[Tuple[Frame, ULID]],
+        pbar: Tqdm,
+        *,
+        skip_uploaded_files: bool,
+    ) -> FrameDataGenerator:
+        if skip_uploaded_files:
+            yield from self._skip_upload_data(source_frames, pbar)
+            return
+
+        for frame, frame_id in source_frames:
+            for sensor_name, data in frame.items():
+                if not isinstance(data, RemoteData):
+                    yield data, sensor_name, frame_id.str
+                else:
+                    pbar.update()
+
+    def _skip_upload_data(
+        self, source_frames: Iterator[Tuple[Frame, ULID]], pbar: Tqdm
+    ) -> FrameDataGenerator:
+        done_frames: Dict[float, Frame] = {
+            frame.frame_id.timestamp().timestamp: frame for frame in self.list_frames()
+        }
+        for frame, frame_id in source_frames:
+            done_frame = done_frames.get(frame_id.timestamp().timestamp)
+            for sensor_name, data in frame.items():
+                if not isinstance(data, RemoteData):
+                    if done_frame:
+                        if done_frame[sensor_name] == data.target_remote_path:
```

Comment:
`done_frame[sensor_name]` should return a `Data`, 

what if `done_frame[sensor_name]` raise `KeyError`